sphinxcontrib/needs/directives/needimport.py

```diff
@@ -52,12 +52,30 @@ def run(self):
         need_import_path = self.arguments[0]
 
         if not os.path.isabs(need_import_path):
-            need_import_path = os.path.join(env.app.confdir, need_import_path)
+            # Relative path should starts from current rst file directory
+            curr_dir = os.path.dirname(self.docname)
+            new_need_import_path = os.path.join(env.app.confdir, curr_dir, need_import_path)
+
+            correct_need_import_path = new_need_import_path
+            if not os.path.exists(new_need_import_path):
+                # Check the old way that calculates relative path starting from conf.py directory
+                old_need_import_path = os.path.join(env.app.confdir, need_import_path)
+                if os.path.exists(old_need_import_path):
+                    correct_need_import_path = old_need_import_path
+                    logger.warning(
+                        "Deprecated needimport relative path calculatation based on directory of conf.py used. "
```

Comment:
Deprecation warning: Relative path must be relative to the current document in future, not to the conf.py location.
Use a starting "/", like "/needs.json", to make the path relative to conf.py.